src/software/ai/hl/stp/play/play.h

```diff
@@ -179,7 +179,7 @@ class Play
 
     // TODO (#2359): remove this
     // The Play's knowledge of the most up-to-date World
-    std::optional<World> world_;
+    std::optional<WorldPtr> world_ptr_;
```

Comment:
Optional pointer is interesting. I think traditionally you would just have the pointer be a nullptr, but I guess this makes sense if we're going for the pattern of a `WorldPtr` object should _never_ be a nullptr and as such we never need to have a nullptr check.